java/runtime/src/main/java/org/ray/runtime/task/LocalModeTaskSubmitter.java

```diff
@@ -152,14 +153,18 @@ private void returnTaskExecutor(TaskExecutor worker, TaskSpec taskSpec) {
       FunctionDescriptor functionDescriptor, List<FunctionArg> args) {
     byte[] taskIdBytes = new byte[TaskId.LENGTH];
     new Random().nextBytes(taskIdBytes);
+    List<String> functionDescriptorList = functionDescriptor.toList();
     return TaskSpec.newBuilder()
         .setType(taskType)
         .setLanguage(Language.JAVA)
         .setJobId(
             ByteString.copyFrom(runtime.getRayConfig().getJobId().getBytes()))
         .setTaskId(ByteString.copyFrom(taskIdBytes))
-        .addAllFunctionDescriptor(functionDescriptor.toList().stream().map(ByteString::copyFromUtf8)
-            .collect(Collectors.toList()))
+        .setFunctionDescriptor(org.ray.runtime.generated.Common.FunctionDescriptor.newBuilder()
+                .setJavaFunctionDescriptor(org.ray.runtime.generated.Common.JavaFunctionDescriptor.newBuilder()
+                        .setClassName(functionDescriptorList.get(0))
+                        .setFunctionName(functionDescriptorList.get(1))
+                        .setSignature(functionDescriptorList.get(2))))
```

Comment:
Predictions.assert that there're 3 items in functionDescriptorList?

Reply:
I added a Preconditions.checkState here.